src/simulation_runner.cu

```diff
@@ -63,9 +63,10 @@ void simulation_runner::run_simulation(stats_composite& stats_runner)
 	// initialize states
 	run_initialize_random(trajectory_batch_limit, seed_, d_rands.get());
 
-	run_initialize_initial_state(trajectory_batch_limit, fixed_initial_part_, free_mask_, d_last_states.get(),
-								 d_last_times.get(), d_rands.get());
-
+	thrust::device_vector<float> d_initial_values = initial_values_;
```

Comment:
can you add this definition 4 lines above so all device allocations are together? :)